mypy/suggestions.py

```diff
@@ -0,0 +1,463 @@
+
+        if give_json:
+            return self.json_suggestion(function, suggestion)
+        else:
+            return suggestion
+
+    def suggest_callsites(self, function: str) -> str:
+        """Find a list of call sites of function."""
+        with self.restore_after(function):
+            _, _, _, node = self.find_node(function)
+            callsites, _ = self.get_callsites(node)
+
+        return '\n'.join(dedup(
+            ["%s:%s: %s" % (path, line, self.format_args(arg_kinds, arg_names, arg_types))
+             for path, line, arg_kinds, _, arg_names, arg_types in callsites]
+        ))
+
+    @contextmanager
+    def restore_after(self, target: str) -> Iterator[None]:
+        """Context manager that reloads a module after executing the body.
+
+        This should undo any damage done to the module state while mucking around.
+        """
+        try:
+            yield
+        finally:
+            module = module_prefix(self.graph, target)
+            if module:
+                self.reload(self.graph[module])
+
+    def get_trivial_type(self, fdef: FuncDef) -> CallableType:
+        """Generate a trivial callable type from a func def, with all Anys"""
+        return CallableType(
+            [AnyType(TypeOfAny.unannotated) for a in fdef.arg_kinds],
+            fdef.arg_kinds,
+            fdef.arg_names,
+            AnyType(TypeOfAny.unannotated),
+            self.builtin_type('builtins.function'))
+
+    def get_args(self, is_method: bool,
+                 base: CallableType, defaults: List[Optional[Type]],
+                 callsites: List[Callsite]) -> List[List[Type]]:
+        """Produce a list of type suggestions for each argument type."""
+        types = []  # type: List[List[Type]]
+        for i in range(len(base.arg_kinds)):
+            # Make self args Any but this will get overriden somewhere in the checker
+            if i == 0 and is_method:
+                types.append([AnyType(TypeOfAny.explicit)])
+                continue
+
+            all_arg_types = []
+            for call in callsites:
+                for typ in call.arg_types[i - is_method]:
+                    # Collect all the types except for explicit anys
+                    if not isinstance(typ, AnyType) or is_explicit_any(typ):
+                        all_arg_types.append(typ)
+            # Add in any default argument types
+            default = defaults[i]
+            if default:
+                all_arg_types.append(default)
+
+            if all_arg_types:
+                types.append(generate_type_combinations(all_arg_types))
+            else:
+                # If we don't have anything, we'll try Any and object
+                types.append([AnyType(TypeOfAny.explicit), self.builtin_type('builtins.object')])
+        return types
+
+    def get_default_arg_types(self, state: State, fdef: FuncDef) -> List[Optional[Type]]:
+        return [state.type_checker().type_map[arg.initializer] if arg.initializer else None
```

Comment:
I fixed the bug that was causing the crash and have kept it as an indexing op. I think it is possible it will need to change to `get` but I want to wait until it is clear that it would be for a real reason and not to work around bugs